receiver/libhoneyreceiver/libhoneyparser.go

```diff
@@ -0,0 +1,129 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package libhoneyreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/libhoneyreceiver"
+
+import (
+	"fmt"
+	"mime"
+	"net/http"
+	"net/url"
+
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	semconv "go.opentelemetry.io/collector/semconv/v1.16.0"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/libhoneyreceiver/internal/simplespan"
+)
+
+func readContentType(resp http.ResponseWriter, req *http.Request) (encoder, bool) {
+	if req.Method != http.MethodPost {
+		handleUnmatchedMethod(resp)
+		return nil, false
+	}
+
+	switch getMimeTypeFromContentType(req.Header.Get("Content-Type")) {
+	case jsonContentType:
+		return jsEncoder, true
+	case "application/x-msgpack", "application/msgpack":
+		return mpEncoder, true
+	default:
+		handleUnmatchedContentType(resp)
+		return nil, false
+	}
+}
+
+func writeResponse(w http.ResponseWriter, contentType string, statusCode int, msg []byte) {
+	w.Header().Set("Content-Type", contentType)
+	w.WriteHeader(statusCode)
+	_, _ = w.Write(msg)
+}
+
+func getMimeTypeFromContentType(contentType string) string {
+	mediatype, _, err := mime.ParseMediaType(contentType)
+	if err != nil {
+		return ""
+	}
+	return mediatype
+}
+
+func handleUnmatchedMethod(resp http.ResponseWriter) {
+	status := http.StatusMethodNotAllowed
+	writeResponse(resp, "text/plain", status, []byte(fmt.Sprintf("%v method not allowed, supported: [POST]", status)))
+}
+
+func handleUnmatchedContentType(resp http.ResponseWriter) {
+	status := http.StatusUnsupportedMediaType
+	writeResponse(resp, "text/plain", status, []byte(fmt.Sprintf("%v unsupported media type, supported: [%s, %s]", status, jsonContentType, pbContentType)))
+}
+
+func getDatasetFromRequest(path string) (string, error) {
+	if path == "" {
+		return "", fmt.Errorf("missing dataset name")
+	}
+	dataset, err := url.PathUnescape(path)
+	if err != nil {
+		return "", err
+	}
+	return dataset, nil
+}
+
+func toPsomething(dataset string, ss []simplespan.SimpleSpan, cfg Config, logger zap.Logger) plog.Logs {
```

Comment:
Lets rename to `toLogs` and make this log specific and then have separate functions for converting to a span. That way we can return `plog.Logs` and `ptrace.Trace` instead of an `any`.

Reply:
Ya toPdata and 2 return values is good if the incoming request batch can contain different signals.